eservice/pdo/eservice/scripts/EServiceCLI.py

```diff
@@ -125,183 +176,34 @@ def render_GET(self, request) :
-
-        except Exception as e :
-            logger.error('unknown exception processing request (Invoke); %s', str(e))
-            return ErrorResponse(request, http.BAD_REQUEST, 'unknown exception processing request')
-
-        try :
-            # result = json.dumps(response).encode()
-
-            request.setResponseCode(http.OK)
-            request.setHeader('content-type', 'application/text')
-            request.write(response.encode())
-
-            return request
-
-        except Exception as e :
-            logger.error("unknown exception packing response (Invoke); %s", str(e))
-            return ErrorResponse(request, http.BAD_REQUEST, "unknown exception while packing response")
-
-    ## -----------------------------------------------------------------
-    def render_POST(self, request) :
-        return self._defer_request_(self._handle_request_, request)
-
-## XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-## XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-class ContractEnclaveServer(Resource):
-
-    ## -----------------------------------------------------------------
-    def __init__(self, enclave, storage_url) :
-        Resource.__init__(self)
-        self.enclave = enclave
-        self.storage_url = storage_url
-
-    ## -----------------------------------------------------------------
-    def getChild(self, name ,request) :
-        logger.debug('request: %s', name)
-        if name == b'invoke' :
-            return InvokeResource(self.enclave)
-        elif name == b'verify' :
-            return VerifyResource(self.enclave)
-        elif name == b'info' :
-            return InfoResource(self.enclave, self.storage_url)
-        elif name == b'shutdown' :
-            return ShutdownResource()
-        else :
-            return NoResource()
-
 # -----------------------------------------------------------------
 # -----------------------------------------------------------------
 def StartEnclaveService(http_host, http_port, enclave, storage_url) :
     logger.info('service started on port %s', http_port)
 
-    root = ContractEnclaveServer(enclave, storage_url)
-    site = Site(root)
+    thread_pool = ThreadPool(maxthreads=8)
+    thread_pool.start()
+    reactor.addSystemEventTrigger('before', 'shutdown', thread_pool.stop)
+
+    root = Resource()
+    root.putChild(b'shutdown', ShutdownResource())
+    root.putChild(b'info', WSGIResource(reactor, thread_pool, InfoApp(enclave, storage_url)))
+    root.putChild(b'invoke', WSGIResource(reactor, thread_pool, InvokeApp(enclave)))
+    root.putChild(b'verify', WSGIResource(reactor, thread_pool, VerifyApp(enclave)))
+
+    # root.putChild(b'invoke', InvokeResource(enclave))
+    # root.putChild(b'verify', VerifyResource(enclave))
+    # root.putChild(b'info', InfoResource(enclave, storage_url))
+
+    site = Site(root, timeout=60)
+    site.displayTracebacks = True
+
+    reactor.suggestThreadPoolSize(10)
```

Comment:
I guess this is my lack of understanding of twisted--- why do we set the threadpoolsize twice (once max at 8, and once 10)?

Reply:
ultimately... there are two different thread pools. one for the WSIG controller and one for the reactor.